From d7ab3b104f1c68100548bba00cbf915dbe69e2a8 Mon Sep 17 00:00:00 2001
From: Josua Mayer <josua@solid-run.com>
Date: Wed, 22 Jul 2020 15:22:48 +0200
Subject: [PATCH 14/26] lx2160a: add ramdisk_addr_r for distro-boot support

Because ramdisks can be huge, the addresses were reordered by size so
that ramdisk is loaded beyond kernel and fdt:
- 0x81000000: 1MB for DTB
- 0x81100000: 255MB for Kernel
- 0x90000000: open end for ramdisk

Signed-off-by: Josua Mayer <josua@solid-run.com>
---
 include/configs/lx2160a_common.h | 7 ++++---
 1 file changed, 4 insertions(+), 3 deletions(-)

diff --git a/include/configs/lx2160a_common.h b/include/configs/lx2160a_common.h
index c5c1dfc4d348..85f36b4cdb17 100644
--- a/include/configs/lx2160a_common.h
+++ b/include/configs/lx2160a_common.h
@@ -254,7 +254,8 @@ int select_i2c_ch_pca9547_sec(unsigned char ch);
 
 #define EXTRA_ENV_SETTINGS			\
 	"hwconfig=fsl_ddr:bank_intlv=auto\0"	\
-	"ramdisk_addr=0x800000\0"		\
+	"ramdisk_addr=0x90000000\0"		\
+	"ramdisk_addr_r=0x90000000\0"		\
 	"ramdisk_size=0x2000000\0"		\
 	"fdt_high=0xa0000000\0"			\
 	"initrd_high=0xffffffffffffffff\0"	\
@@ -265,9 +266,9 @@ int select_i2c_ch_pca9547_sec(unsigned char ch);
 	"scripthdraddr=0x80080000\0"		\
 	"fdtheader_addr_r=0x80100000\0"		\
 	"kernelheader_addr_r=0x80200000\0"	\
-	"kernel_addr_r=0x81000000\0"		\
+	"kernel_addr_r=0x81100000\0"		\
 	"kernelheader_size=0x40000\0"		\
-	"fdt_addr_r=0x90000000\0"		\
+	"fdt_addr_r=0x81000000\0"		\
 	"load_addr=0xa0000000\0"		\
 	"kernel_size=0x2800000\0"		\
 	"kernel_addr_sd=0x8000\0"		\
-- 
2.30.2

